pkg/cmd/cli/cmd/importimage.go

```diff
@@ -0,0 +1,139 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+	"github.com/openshift/origin/pkg/cmd/cli/describe"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	importImageLong = `TODO`
+
+	importImageExample = `  // TODO
+  $ %[1]s import-image mystream`
+)
+
+// NewCmdImportImage implements the OpenShift cli import-image command.
+func NewCmdImportImage(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "import-image IMAGESTREAM",
+		Short:   "TODO.",
+		Long:    importImageLong,
+		Example: fmt.Sprintf(importImageExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunImportImage(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	//cmd.Flags().Bool("dump-logs", false, "Specify if the build logs for the cancelled build should be shown.")
+	//cmd.Flags().Bool("restart", false, "Specify if a new build should be created after the current build is cancelled.")
+	return cmd
+}
+
+// RunImportImage contains all the necessary functionality for the OpenShift cli import-image command.
+func RunImportImage(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 || len(args[0]) == 0 {
+		return cmdutil.UsageError(cmd, "You must specify the name of an image stream.")
+	}
+
+	streamName := args[0]
+	namespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	client, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	imageStreamClient := client.ImageStreams(namespace)
+	stream, err := imageStreamClient.Get(streamName)
+	if err != nil {
+		return err
+	}
+
+	if len(stream.Spec.DockerImageRepository) == 0 {
```

Comment:
The image importer imports all tags and doesn't currently look at spec.tags. What behavior do you want?